src/lib/rc/dsm.cpp

```diff
@@ -555,7 +661,7 @@ dsm_decode(hrt_abstime frame_time, uint16_t *values, uint16_t *num_values, bool
 	/* check all values */
 	for (unsigned i = 0; i < *num_values; i++) {
 		/* if the value is unrealistic, fail the parsing entirely */
-		if (values[i] < 600 || values[i] > 2400) {
+		if (values[i] < 897 || values[i] > 2103) {
```

Comment:
Because they are 3ms beyond normal limits I get exactly what you did here.  I just think it will be hard for future coders to follow without something like a `static const int PWM_BUFFER {3};`

Reply:
Actually it's a bit weirder, these are the effective extremes with the decoded raw data (1024 or 2048 bits) scaled with our integer scaling above (https://github.com/PX4/Firmware/pull/15430/files#diff-53b65610c058ef465c0049721d814061R609). 

Thinking about this again the full range doesn't really make sense because we extract the full range from the packet (2048 bits), then shift + scale it. So the only way it's going to be outside of this range is if it slipped through parsing at 0. I'll take another look and likely drop these magic numbers entirely.

Reply:
I updated the scaling slightly to match the specification and added a comment.